### PR TITLE
Yang la1321.cement china

### PR DESCRIPTION
**Pull request by thuliuyang**

### CODE REVIEW

**Comment by rynacui** [comment]
@bpbond The appveyor test still has problem finding the `gcamdata.compdata` package for all the PRs we have now. Do you have time to take a look? Thank you!

**Comment by bpbond** [comment]
Hmm, for some reason it's dying with an out-of-memory error:
```
198Downloading GitHub repo JGCRI/gcamdata.compdata@gcamdata-China
199Installing package into 'c:/RLibrary'
200(as 'lib' is unspecified)
201* installing *source* package 'gcamdata.compdata' ...
202** R
203** data
204** byte-compile and prepare package for lazy loading
205** help
206*** installing help indices
207  converting help for package 'gcamdata.compdata'
208    finding HTML links ... done
209    COMPDATA                                html  
210** building package indices
211Error : cannot allocate vector of size 1.9 Mb
212ERROR: installing package indices failed
213* removing 'c:/RLibrary/gcamdata.compdata'
```

**Comment by bpbond** [comment]
I'm not sure why that Travis build is showing as pending–it's passed. Go ahead and merge this when ready @rynacui .

**Comment by rynacui** [comment]
Thanks a lot @bpbond! It's pending on @sypnl's review.

File: R/zchunk_LA1321.Cement_china.R [hunk 1]
```diff
@@ -0,0 +1,117 @@
+#' module_gcam.china_LA1321.Cement
+#'
+#' Allocate across the provinces national cement production, input-output cofficients, and energy inputs to cement production
```
Comment (bpbond) [inline R/zchunk_LA1321.Cement_china.R]: This is unclear because of confusing grammar.

File: R/zchunk_LA1321.Cement_china.R [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+#'
+#' @param command API command to execute
+#' @param ... other optional parameters, depending on command
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{L1321.out_Mt_province_cement_Yh}, \code{L1321.IO_GJkg_province_cement_F_Yh}, \code{L1321.in_EJ_province_cement_F_Y}. The corresponding file in the
+#' original data system was \code{LA1321.Cement.R} (gcam-china level1).
+#' @details The tables for cement production, i.e., out, and energy inputs, i.e., in, were calculated by applying state shares to national data.
+#' @details The state shares were determined by the states' relative values of cement shipments.
+#' @details The input-out coefficients were downscaled to the states in proportation to the national data.
```
Comment (bpbond) [inline R/zchunk_LA1321.Cement_china.R]: "proportion"

File: R/zchunk_LA1321.Cement_china.R [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter mutate select
+#' @importFrom tidyr gather spread
+#' @author Yang Aug 2018
```
Comment (bpbond) [inline R/zchunk_LA1321.Cement_china.R]: Please use full name or multiple initials.

File: R/zchunk_LA1321.Cement_china.R [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+    sector <- fuel <- province <- multiplier <- GCAM_region_ID <- year <- value <- NULL   # silence package check notes
+
+    # Load required inputs
+    L1321.out_Mt_R_cement_Yh <- get_data(all_data, "L1321.out_Mt_R_cement_Yh")
+    L1321.IO_GJkg_R_cement_F_Yh <- get_data(all_data, "L1321.IO_GJkg_R_cement_F_Yh")
+    L1321.in_EJ_R_cement_F_Y <- get_data(all_data, "L1321.in_EJ_R_cement_F_Y")
+
+    # Proxy for downscaling cement to provinces
+    # TODO: Find a better proxy
```
Comment (bpbond) [inline R/zchunk_LA1321.Cement_china.R]: Please move this to a repo issue and delete this comment (or if you want to keep, reference the issue number).

File: R/zchunk_LA1321.Cement_china.R [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    L101.inNBS_Mtce_province_S_F <- get_data(all_data, "L101.inNBS_Mtce_province_S_F")
+
+    # ===================================================
+    # Assigning national cement production to provinces
+    L101.inNBS_Mtce_province_S_F %>%
+      filter(sector == "industry" & fuel == "coal") %>%
+      group_by(year) %>%
+      mutate(multiplier = value /sum(value)) %>%
```
Comment (bpbond) [inline R/zchunk_LA1321.Cement_china.R]: `value / sum`

File: R/zchunk_LA1321.Cement_china.R [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+      ungroup %>%
```
Comment (bpbond) [inline R/zchunk_LA1321.Cement_china.R]: Not needed since you're about to immediately `group_by` again

File: R/zchunk_LA1321.Cement_china.R [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+      group_by(province) %>%
+      mutate(multiplier = approx_fun(year, multiplier, rule = 2), sector = "cement") %>%
+      ungroup() %>%
+      select(-value, -fuel) ->
+      L1321.in_pct_province_cement
+
+    # This section is calculating province-level data by multiplying the province share by the china component in the global data
+    # This will generate an output table: Cement production by province / historical year
```
Comment (bpbond) [inline R/zchunk_LA1321.Cement_china.R]: Good, clear comment--thanks!

File: R/zchunk_LA1321.Cement_china.R [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    L1321.in_pct_province_cement %>%
+      left_join_error_no_match(L1321.out_Mt_R_cement_Yh %>% filter(GCAM_region_ID == 11), by = c("year", "sector")) %>%
```
Comment (bpbond) [inline R/zchunk_LA1321.Cement_china.R]: Here and throughout, let's use a constant `gcam.CHINA_CODE` defined in `constants.R` please, not a numeric code. (See lines 35-36 in that file.)

File: R/zchunk_LA1321.Cement_china.R [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+      mutate(value = value * multiplier) %>% # Multiplying the national amount with the province share
+      select(province, sector, year, value) ->
+      L1321.out_Mt_province_cement_Yh
+
+    # This section is downscaling the national input/output (IO) coefficients to the province level
+    # Assuming all provinces have the same IO coefficients for heat, electricity, and limestone
+    # This will generate an output table: Input-output coefficients of cement production by province / fuel / historical year
+    L1321.IO_GJkg_R_cement_F_Yh %>%
+      filter(GCAM_region_ID == 11) %>%
+      repeat_add_columns(tibble(province = unique( L1321.out_Mt_province_cement_Yh$province))) %>%
```
Comment (bpbond) [inline R/zchunk_LA1321.Cement_china.R]: `unique(L1321`

File: R/zchunk_LA1321.Cement_china.R [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+#' Allocate across the provinces national cement production, input-output cofficients, and energy inputs to cement production
+#'
+#' @param command API command to execute
+#' @param ... other optional parameters, depending on command
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{L1321.out_Mt_province_cement_Yh}, \code{L1321.IO_GJkg_province_cement_F_Yh}, \code{L1321.in_EJ_province_cement_F_Y}. The corresponding file in the
+#' original data system was \code{LA1321.Cement.R} (gcam-china level1).
+#' @details The tables for cement production, i.e., out, and energy inputs, i.e., in, were calculated by applying state shares to national data.
+#' @details The state shares were determined by the states' relative values of cement shipments.
```
Comment (sypnl) [inline R/zchunk_LA1321.Cement_china.R]: "provincial shares" & "provinces' "

File: R/zchunk_LA1321.Cement_china.R [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+    all_data <- list(...)[[1]]
+
+    sector <- fuel <- province <- multiplier <- GCAM_region_ID <- year <- value <- NULL   # silence package check notes
+
+    # Load required inputs
+    L1321.out_Mt_R_cement_Yh <- get_data(all_data, "L1321.out_Mt_R_cement_Yh")
+    L1321.IO_GJkg_R_cement_F_Yh <- get_data(all_data, "L1321.IO_GJkg_R_cement_F_Yh")
+    L1321.in_EJ_R_cement_F_Y <- get_data(all_data, "L1321.in_EJ_R_cement_F_Y")
+
+    # Proxy for downscaling cement to provinces
```
Comment (sypnl) [inline R/zchunk_LA1321.Cement_china.R]: Clarify what the current proxy is

**Pull request merged**